contentcuration/contentcuration/frontend/shared/app.js

```diff
@@ -43,20 +44,31 @@ Vue.component('ActionLink', ActionLink);
 
 export let rootVue;
 
-export default function startApp({ store, router, index }) {
-  initializeDB().then(() => {
-    const config = {
-      el: 'app',
-      store,
-      router,
-    };
-
-    if (index) {
-      Object.assign(config, index);
-    } else {
-      Object.assign(config, Base);
-    }
-
-    rootVue = new Vue(config);
-  });
+export default async function startApp({ store, router, index }) {
+  await initializeDB();
+
+  const userId = window.user ? window.user.id : null;
+  const isAnonymous = userId === null;
+
```

Comment:
I am not sure how to handle errors for database Promises on lines 53 through 60. I was thinking about calling `resetDB()` whenever anything goes wrong here and logging the error through Raven but I am not sure if it's worth it, these calls don't look so problematic. What's your general strategy?